code/modules/antagonists/blood_worm/actions/inject_blood.dm

```diff
@@ -0,0 +1,243 @@
+
+	var/mob/living/basic/blood_worm/worm = target
+
+	if (worm.health - health_cost < minimum_health)
+		if (feedback)
+			owner.balloon_alert(owner, "out of blood!")
+		return FALSE
+
+	return ..()
+
+/datum/action/cooldown/mob_cooldown/blood_worm/inject/Activate(atom/target)
+	var/mob/living/basic/blood_worm/worm = src.target
+	var/mob/living/carbon/human/host = worm.host
+
+	host.apply_status_effect(status_effect_type, worm)
+
+	host.visible_message(
+		message = span_danger("[host]'s wounds start healing unnaturally quickly!"),
+		ignored_mobs = owner
+	)
+
+	to_chat(owner, span_notice("You inject blood into the damaged tissues of your host."))
+
+	host.blood_volume -= health_cost * BLOOD_WORM_HEALTH_TO_BLOOD
+
+	return ..()
+
+/datum/status_effect/blood_worm_transfuse
+	id = "blood_worm_transfuse"
+	duration = 20 SECONDS
+	show_duration = TRUE
+	alert_type = /atom/movable/screen/alert/status_effect/blood_worm_transfuse
+	status_type = STATUS_EFFECT_REPLACE
+	processing_speed = STATUS_EFFECT_PRIORITY
+
+	var/damage_regen_rate = 0
+
+	var/wound_regen_rate = 0
+	var/wound_regen_accumulation = 0
+
+	var/organ_regen_rate = 0
+
+	var/trauma_regen_rate = 0
+	var/trauma_regen_accumulation = 0
+
+	var/atom/movable/screen/alert/status_effect/worm_alert = null
+	var/mob/living/basic/blood_worm/worm = null
+
+	var/last_tick_time = 0
+
+/datum/status_effect/blood_worm_transfuse/on_creation(mob/living/new_owner, mob/living/basic/blood_worm/new_worm)
+	. = ..()
+	if (!.)
+		return
+
+	attach_to_worm(new_worm)
+	last_tick_time = world.time
+
+/datum/status_effect/blood_worm_transfuse/Destroy()
+	detach_from_worm()
+	return ..()
+
+/datum/status_effect/blood_worm_transfuse/tick(seconds_between_ticks)
+	var/need_mob_update = FALSE
+
+	need_mob_update |= heal_damage(seconds_between_ticks)
+	need_mob_update |= heal_wounds(seconds_between_ticks)
+	need_mob_update |= heal_organs(seconds_between_ticks)
+	heal_traumas(seconds_between_ticks)
+
+	if (need_mob_update)
+		owner.updatehealth()
+
+	last_tick_time = world.time
+
+/datum/status_effect/blood_worm_transfuse/update_shown_duration()
+	. = ..()
+
+	if (worm_alert && linked_alert)
+		worm_alert.maptext = linked_alert.maptext
+
+/datum/status_effect/blood_worm_transfuse/proc/heal_damage(seconds_between_ticks)
+	var/healing_left = damage_regen_rate * seconds_between_ticks
+
+	if (owner.getBruteLoss() > 0 && healing_left > 0)
+		var/amount_healed = max(0, owner.adjustBruteLoss(-healing_left, forced = TRUE, updating_health = FALSE))
+		healing_left -= amount_healed
+		. |= amount_healed
+
+	if (owner.getFireLoss() > 0 && healing_left > 0)
+		var/amount_healed = max(0, owner.adjustFireLoss(-healing_left, forced = TRUE, updating_health = FALSE))
+		healing_left -= amount_healed
+		. |= amount_healed
+
+	if (owner.getToxLoss() > 0 && healing_left > 0)
+		var/amount_healed = max(0, owner.adjustToxLoss(-healing_left, forced = TRUE, updating_health = FALSE))
+		healing_left -= amount_healed
+		. |= amount_healed
```

Comment:
why not just use heal_overall_damage here

Reply:
Because that adjusts brute, burn and stamina damage. I'm adjusting brute, burn and toxin damage.